scripts/aiven/clouds.py

```diff
@@ -1,24 +1,54 @@
-import requests
 import argparse
-from typing import Dict
+import re
+import requests
+from dataclasses import dataclass
+from typing import Dict, Self
+from natsort import natsorted
+
+
+@dataclass
+class CloudEntry:
+    description: str
+    geo_region: str
+    name: str
+    vendor_code: str
+    vendor_name: str
 
+    @classmethod
+    def from_dict(cls: type[Self], cloud: dict[str, str | float], /) -> Self:
+        """Create cloud entry from dict
 
-def create_cloud_entry(cloud: Dict) -> str:
-    """Creates cloud entry with formatted info.
+        :param cloud: contains relevant info about cloud
+        :rtype: CloudEntry
+        """
 
-    :param cloud: contains relevant info about cloud
-    :returns: formatted string with cloud info
-    :rtype: str
-    """
-    entry = ""
-    # Printing in title case to make it look better
-    entry += f'  * - {cloud["geo_region"].title()}'
-    entry += "\n"
-    entry += f'    - ``{cloud["cloud_name"]}``'
-    entry += "\n"
-    prefix = cloud["cloud_description"][0 : cloud["cloud_description"].find("-")]
-    entry += f"    - {prefix}"
-    return entry
+        description_parts = [
+            description_part.strip()
+            for description_part in re.split(r"[,:-]", cloud["cloud_description"])
+        ]
+        vendor_name = description_parts.pop(2)
+        description = f"{description_parts[0]}, {description_parts[1]}: {description_parts[2]}"
+        cloud_name = cloud["cloud_name"]
+        vendor_code = cloud_name[0:cloud_name.find("-")]
```

Comment:
nit: I prefer `index` instead of `find` in such cases otherwise you silently discard the last character if somehow `-` is not part of the string.